src/state/lists/updater.ts

```diff
@@ -1,41 +1,27 @@
-import { useCallback, useEffect } from 'react'
+import { useEffect } from 'react'
 import { useDispatch } from 'react-redux'
 
-import { useWeb3React } from 'hooks'
-import { useFetchListCallback } from 'hooks/useFetchListCallback'
-import useInterval from 'hooks/useInterval'
-import useIsWindowVisible from 'hooks/useIsWindowVisible'
-import { AppDispatch } from 'state/index'
-import { useAllLists } from 'state/lists/hooks'
+import { NETWORKS_INFO, SUPPORTED_NETWORKS } from 'constants/networks'
+import { AppDispatch } from 'state'
+import { getTokenList } from 'utils/getTokenList'
+
+import { setTokenList } from './actions'
 
 export default function Updater(): null {
-  const { library } = useWeb3React()
   const dispatch = useDispatch<AppDispatch>()
-  const isWindowVisible = useIsWindowVisible()
-
-  // get all loaded lists, and the active urls
-  const lists = useAllLists()
-
-  const fetchList = useFetchListCallback()
-  const fetchAllListsCallback = useCallback(() => {
-    if (!isWindowVisible) return
-    Object.keys(lists).forEach(url =>
-      fetchList(url).catch(error => console.debug('interval list fetching error', error)),
-    )
-  }, [fetchList, isWindowVisible, lists])
-
-  // fetch all lists every 10 minutes, but only after we initialize library
-  useInterval(fetchAllListsCallback, library ? 1000 * 60 * 10 : null)
 
-  // whenever a list is not loaded and not loading, try again to load it
   useEffect(() => {
-    Object.keys(lists).forEach(listUrl => {
-      const list = lists[listUrl]
-      if (!list.current && !list.loadingRequestId && !list.error) {
-        fetchList(listUrl).catch(error => console.debug('list added fetching error', error))
-      }
+    SUPPORTED_NETWORKS.forEach(chainId => {
```

Comment:
This will make quite a lot of requests at a moment and just one of the results is needed. Maybe change?

Reply:
updated using MAINNET_NETWORKS. 
I think we need another pr update: call get token on demand. It's currently get all chain. 🐯

Reply:
Reasonable